### PR TITLE
Add return types on public classes and interfaces

### PR DESCRIPTION
**Pull request by GromNaN**
Add return types to methods from the public API.

Follow-up https://github.com/doctrine/data-fixtures/pull/455#pullrequestreview-1782094062

### CODE REVIEW

**Comment by GromNaN** [comment]
I'll check why `ObjectManager::getClassMetadata()` return `null` for `Doctrine\Tests\Common\DataFixtures\TestEntity\Role`.

File: tests/Common/DataFixtures/Executor/ORMExecutorTest.php [hunk 1]
```diff
@@ -52,8 +52,7 @@ public function testExecuteWithPurge(): void
         $em     = $this->getMockSqliteEntityManager();
         $purger = $this->getMockPurger();
         $purger->expects($this->once())
-            ->method('purge')
-            ->willReturn(null);
+            ->method('purge');
```
Comment (GromNaN) [inline tests/Common/DataFixtures/Executor/ORMExecutorTest.php]: Updated because `purge()` return type is `void`.

**Comment by GromNaN** [comment]
Psalm baseline updated. I was confused by the fact that both psalm and phpstan are used.

**Review by greg0ire — APPROVED** [review]

File: src/ReferenceRepository.php [hunk 2]
```diff
@@ -75,7 +75,7 @@ public function __construct(ObjectManager $manager)
      *
      * @return array
```
Comment (derrabus) [inline src/ReferenceRepository.php]: Remove.

File: tests/Common/DataFixtures/Executor/ORMExecutorTest.php (hunk 1 above)
Comment (derrabus) [reply tests/Common/DataFixtures/Executor/ORMExecutorTest.php]: We should backport that change to 1.8.x.

File: tests/Common/DataFixtures/ReferenceRepositoryTest.php [hunk 3]
```diff
@@ -198,6 +198,9 @@ public function testGetIdentifierWhenHasNotBeenManagedYetByUnitOfWork(): void
             ->method('getIdentifierValues')
             ->with($role)
             ->willReturn($identitiesExpected);
+        $classMetadata->expects($this->once())
+            ->method('getName')
+            ->willReturn(Role::class);
```
Comment (derrabus) [inline tests/Common/DataFixtures/ReferenceRepositoryTest.php]: Why did this change become necessary? Should we backport?

Comment (GromNaN) [reply tests/Common/DataFixtures/ReferenceRepositoryTest.php]: Yes: https://github.com/doctrine/data-fixtures/pull/463

**Comment by derrabus** [comment]
Rebase needed.

File: src/FixtureInterface.php [hunk 4]
```diff
@@ -14,5 +14,5 @@ interface FixtureInterface
     /**
      * Load data fixtures with the passed EntityManager
      */
-    public function load(ObjectManager $manager);
+    public function load(ObjectManager $manager): void;
```
Comment (GromNaN) [inline src/FixtureInterface.php:17]: This change can impact every project using data-fixtures. Adding the `@return` type  annotation in the previous version will ease migration: #472

**Comment by derrabus** [comment]
Can we enable this rule now?

https://github.com/doctrine/data-fixtures/blob/36d20789751e0fdef7540336d0fc0a69d6e07fa1/phpcs.xml.dist#L59-L62

File: phpcs.xml.dist [hunk 5]
```diff
@@ -56,11 +56,6 @@
         <exclude-pattern>src/*</exclude-pattern>
     </rule>
 
-    <rule ref="SlevomatCodingStandard.TypeHints.ReturnTypeHint.MissingNativeTypeHint">
```
Comment (GromNaN) [inline phpcs.xml.dist:59]: Response to https://github.com/doctrine/data-fixtures/pull/459#issuecomment-1991384911: This rule is applied by default. This config was to exclude `src` dir.

Comment (derrabus) [reply phpcs.xml.dist:59]: Right. I missed that you've changed the config already. 🙈

**Review by derrabus — APPROVED** [review]

**Pull request merged**